### PR TITLE
typescript: init typescript

### PR DESCRIPTION
**Pull request by sorja**
PR Objective:
- [x] Setup initial configuration for typescript
- [x] Convert `webapp/main.js` to working `webapp/main.tsx`
- [x] Configure tslint
- ...testing?

### CODE REVIEW

File: webapp/main.tsx [hunk 1]
```diff
@@ -3,15 +3,15 @@ import '@webapp/components/ckEditor/style.less'
...
+import * as React from 'react'
+import * as ReactDOM from 'react-dom'
 import { Provider } from 'react-redux'
 import { BrowserRouter } from 'react-router-dom'
 
 import PageRoutes from './pages/PageRoutes'
 import store from './main/store'
 
-function renderApp() {
+const App = () => {
```
Comment (minotogna) [inline webapp/main.tsx:14]: let's rename this to Main

File: tslint.json [hunk 2]
```diff
@@ -0,0 +1,4 @@
+{
```
Comment (minotogna) [inline tslint.json:1]: if we have tslint, should we remove eslint?

Comment (sorja) [reply tslint.json:1]: We can keep both for now as the project has both plain javascript and will incrementally grow in typescript:
> When it comes to linting TypeScript code, there are two major linting options to choose from: TSLint and ESLint. TSLint is a linter that can only be used for TypeScript, while ESLint supports both JavaScript and TypeScript.

File: webapp/main.tsx [hunk 3]
```diff
@@ -3,15 +3,15 @@ import '@webapp/components/ckEditor/style.less'
 
 import '@webapp/utils/polyfill/polyfill'
 
-import React from 'react'
-import ReactDOM from 'react-dom'
+import * as React from 'react'
+import * as ReactDOM from 'react-dom'
```
Comment (minotogna) [inline webapp/main.tsx:7]: do not use import *

**Review by minotogna — CHANGES_REQUESTED** [review]

**Review by minotogna — APPROVED** [review]

**Pull request merged**